presto-parquet/src/main/java/com/facebook/presto/parquet/reader/ParquetReader.java

```diff
@@ -206,7 +209,18 @@ private ColumnChunk readPrimitive(PrimitiveField field)
             ParquetColumnChunk columnChunk = new ParquetColumnChunk(descriptor, buffer, 0);
             columnReader.setPageReader(columnChunk.readAllPages());
         }
-        return columnReader.readPrimitive(field);
+        try {
+            return columnReader.readPrimitive(field);
+        }
+        catch (UnsupportedOperationException e) {
```

Comment:
Instead of relying on the `UnsupportedOperationException` thrown by the Type classes, it may be cleaner to throw a specific exception (e.g., `ParquetSchemaMismatchException` or sth like that) in `PrimitiveColumnReader::readValue()`. What do you think?